api/src/pcapi/core/offers/api.py

```diff
@@ -1265,6 +1265,7 @@ def delete_unwanted_existing_product(idAtProviders: str) -> None:
         product.isGcuCompatible = False
         product.isSynchronizationCompatible = False
         repository.save(product)
+        search.async_index_offer_ids([offer.id for offer in offers.all()])
```

Comment:
(détail / performance) `offers.all()` va récupérer toutes les colonnes de la table `offer` pour construire des objets `Offer` complets. En fait, ici, on n'a besoin que des identifiants, donc on peut alléger la requête : 

```suggestion
        offer_ids = [id_ for id_, in offers.with_entities(models.Offer.Id)]
        search.async_index_offer_ids(offer_ids)
```

Reply:
Question de noob, a quoi sert le suffix underscore dans `id_` ? Je le vois aussi souvent dans les méthodes sql alchemy

Reply:
C'est pour ne pas "confondre" avec (et écraser) la fonction Python [id()](https://docs.python.org/3/library/functions.html#id). En pratique, on n'utilise que très rarement `id()`. Et ici on ne l'utilise pas, donc on pourrait très bien nommer la variable `id`. C'est plus une convention de ne pas écraser ce genre de fonctions built-in (cf. [liste ici](https://docs.python.org/3/library/functions.html)). Autres cas similaires courants : `type`, `input`, `class`, etc. qu'on nomme souvent avec un `_` final (on trouve aussi parfois `klass` au lieu de `class_`).